### PR TITLE
Added MonthPlausibilityComparator #14457

### LINKED ISSUE #14457: Add MonthPlausibilityComparator
Have

```bibtex
@Article{Jr.2025,
  month      = {Jun},
}
```

and

```bibtex
@Article{Jr.2025,
  month         = jun,
}
```

The latter is better. The merge dialog should select the second column. -- See https://github.com/JabRef/jabref/issues/14454 for a similar scenario (other field)

Task:

1. Create `org.jabref.logic.bibtex.comparator.plausibility.MonthPlausibilityComparator`
2. Wire it into `org.jabref.logic.bibtex.comparator.plausibility.MonthComparatorFactory`. Use `FieldProperty.MONTH`
3. Add test cases

---

Hints:

- Do test-driven development.
- Look at the failed PR https://github.com/JabRef/jabref/pull/14629. It is 90% done, you can base on this.

### PR DESCRIPTION
**Pull request by keerthansamudrala**
### **User description**
Closes #14457

I have implemented the MonthPlausibilityComparator to handle month field reconciliation. This comparator uses a priority-based hierarchy (JabRef format > BibTeX short name > Numbers) to suggest the most standard month format when merging or comparing entries.

**Note to Reviewers**: I have included temporary comments within the logic to explain the specific decision-making for each "Tier" (JabRef format, Short names, Numbers, etc.). If the logic and implementation meet the project standards, I will remove these comments to provide a final "clean code" version for merging.

### Steps to test

**1. Run the newly added JUnit test :** 
        org.jabref.logic.bibtex.comparator.plausibility.MonthPlausibilityComparatorTest.

**2. Verify that the 16 tests pass, covering scenarios such as:**
        - Preference for #jan# over jan.
        - Preference for jan over January.
        - Preference for 1 over January.
        - Correct handling of conflicting months (e.g., June vs. July) by returning UNDETERMINED.
Screenshot 2025-12-28 at 3 34 00 PM

### Mandatory checks

- [x] I own the copyright of the code submitted and I license it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] I manually tested my changes in running JabRef (always required)
- [x] I added JUnit tests for changes (if applicable)
- [x] I added screenshots in the PR description (if change is visible to the user)
- [ ] I described the change in `CHANGELOG.md` in a way that is understandable for the average user (if change is visible to the user)
- [ ] I checked the [user documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request updating file(s) in <https://github.com/JabRef/user-documentation/tree/main/en>.

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Implemented MonthPlausibilityComparator for month field reconciliation

- Uses priority-based hierarchy: JabRef format > short names > numbers

- Added 16 comprehensive JUnit tests covering various month format scenarios

- Integrated comparator into PlausibilityComparatorFactory for automatic field handling

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Month Field Values"] -- "parse" --> B["MonthPlausibilityComparator"]
  B -- "priority hierarchy" --> C["JabRef Format"]
  B -- "priority hierarchy" --> D["Short Names"]
  B -- "priority hierarchy" --> E["Numbers"]
  B -- "priority hierarchy" --> F["String Length"]
  C --> G["ComparisonResult"]
  D --> G
  E --> G
  F --> G
  H["PlausibilityComparatorFactory"] -- "register" --> B
```

<details><summary><h3>File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>MonthPlausibilityComparator.java</strong><dd><code>New MonthPlausibilityComparator implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

jablib/src/main/java/org/jabref/logic/bibtex/comparator/plausibility/MonthPlausibilityComparator.java

<ul><li>New comparator implementing FieldValuePlausibilityComparator interface<br> <li> Implements priority-based month format comparison: JabRef format <br>(#jan#) > short names (jan) > numbers (1) > string length<br> <li> Handles invalid month values and conflicting months by returning <br>UNDETERMINED<br> <li> Uses Month.parse() to validate and normalize month values</ul>

</details>

  </td>
  <td><a href="https://github.com/JabRef/jabref/pull/14745/files#diff-d6ddf06673fd7fb8fbf653edeb15e2f92ddea2e52831af19ddbe03685f8b9cae">+79/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>PlausibilityComparatorFactory.java</strong><dd><code>Register MonthPlausibilityComparator in factory</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

jablib/src/main/java/org/jabref/logic/bibtex/comparator/plausibility/PlausibilityComparatorFactory.java

<ul><li>Added registration of MonthPlausibilityComparator for fields with <br>FieldProperty.MONTH<br> <li> Integrated new comparator into factory's getPlausibilityComparator <br>method<br> <li> Follows existing pattern for other field type comparators</ul>

</details>

  </td>
  <td><a href="https://github.com/JabRef/jabref/pull/14745/files#diff-367fba44bbf181cab141678a073b646f50b9766d8b2c1f45993043301c82cd75">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>MonthPlausibilityComparatorTest.java</strong><dd><code>Comprehensive test suite for MonthPlausibilityComparator</code>&nbsp; </dd></summary>
<hr>

jablib/src/test/java/org/jabref/logic/bibtex/comparator/plausibility/MonthPlausibilityComparatorTest.java

<ul><li>New parameterized JUnit test class with 16 test cases<br> <li> Tests preference hierarchy: JabRef format over short names, short <br>names over numbers, numbers over full names<br> <li> Tests edge cases: invalid months, empty strings, conflicting months, <br>case sensitivity<br> <li> Validates correct handling of UNDETERMINED results for conflicting or <br>invalid inputs</ul>

</details>

  </td>
  <td><a href="https://github.com/JabRef/jabref/pull/14745/files#diff-35357d857d3fd2c3cfde5078063d7d8e10e8c7519be26a515e35c183e9c4e8a0">+18/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Formatting</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>EntryTypePlausibilityComparator.java</strong><dd><code>Reformat JavaDoc comment formatting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

jablib/src/main/java/org/jabref/logic/bibtex/comparator/plausibility/EntryTypePlausibilityComparator.java

<ul><li>Reformatted JavaDoc comment from single line to multi-line format<br> <li> No functional changes to the comparator logic</ul>

</details>

  </td>
  <td><a href="https://github.com/JabRef/jabref/pull/14745/files#diff-876cf0398e38a1df7fa0404cc24451630923195c22c85971129d608826861a96">+13/-1</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Project.xml</strong><dd><code>Update IDE code style configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.idea/codeStyles/Project.xml

<ul><li>Removed ENABLE_JAVADOC_FORMATTING option setting<br> <li> Added WRAP_COMMENTS option set to true for JAVA code style<br> <li> IDE code style configuration updates</ul>

</details>

  </td>
  <td><a href="https://github.com/JabRef/jabref/pull/14745/files#diff-11745b49c51a1ecde7d4d4cffda53d76594e6711ec81bf1148a05942b0438141">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by keerthansamudrala** [comment]
Hi,

I converted the comment to a Javadoc-style block and ran IntelliJ’s reformat (⌘⌥L) with standard options (optimize imports, rearrange code, code cleanup).

IntelliJ reports “No lines changed: content is already properly formatted”, and I consistently end up with the current layout of the Javadoc comment.

Could you please confirm whether this formatting is acceptable and aligns with JabRef’s preferred style, or if you would prefer a different comment structure (for example, a single-line Javadoc or a different placement)?

Thanks!
Screenshot 2026-01-21 at 8 28 16 PM

File: jablib/src/main/java/org/jabref/logic/bibtex/comparator/plausibility/EntryTypePlausibilityComparator.java [hunk 1]
```diff
@@ -3,7 +3,19 @@
 import org.jabref.logic.bibtex.comparator.ComparisonResult;
 import org.jabref.model.entry.types.StandardEntryType;
 
-/// If the left entry type is misc then prefer the the right value 
+/// If
+/// the
```
Comment (Siedlerchr) [inline jablib/src/main/java/org/jabref/logic/bibtex/comparator/plausibility/EntryTypePlausibilityComparator.java]: rever the javadoc formatting here

**Comment by koppor** [comment]
> Could you please confirm whether this formatting is acceptable and aligns with JabRef’s preferred style, o

You can research why IntelliJ does that line breaking... And search for IntelliJ issues... We saw this behavior but did not track it down to the root cause yet.
